Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1475,9 +1497,21 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
         accumulateIntrinsicSizes(gridItem.get());
     }
 
-    for (auto it = masonryIndefiniteItems.begin(), end = masonryIndefiniteItems.end(); it != end; ++it) {
-        CheckedPtr<RenderBox> gridItem = it.get();
-        accumulateIntrinsicSizes(gridItem.get());
+    if (m_renderGrid->isMasonry()) {
+        for (auto& gridItem : masonryIndefiniteItems.indefiniteItems)
+            accumulateIntrinsicSizes(&gridItem);
+
+        /* Optimization for non flex tracks:
+        * If we are in a non flex track we can then skip having to evaluate every item.
+        * 
+        * FIXME: Add support for optimizing the flex track scenario.
+        */
```

Comment:
While I did not see it in our style guide, it looks like the rest of the code uses `//` even for comments over multiple lines